pkg/ctl/cmdutils/nodegroup_filter.go

```diff
@@ -0,0 +1,134 @@
+package cmdutils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gobwas/glob"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha4"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+)
+
+// NodeGroupFilter holds filter configuration
+type NodeGroupFilter struct {
+	IgnoreAllExisting bool
+
+	existing sets.String
+	only     []glob.Glob
+	onlySpec string
+}
+
+// NewNodeGroupFilter create new NodeGroupFilter instance
+func NewNodeGroupFilter() *NodeGroupFilter {
+	return &NodeGroupFilter{
+		IgnoreAllExisting: true,
+
+		existing: sets.NewString(),
+	}
+}
+
+// ApplyOnlyFilter parses given globs for exclusive filtering
+func (f *NodeGroupFilter) ApplyOnlyFilter(globExprs []string, cfg *api.ClusterConfig) error {
+	for _, expr := range globExprs {
+		compiledExpr, err := glob.Compile(expr)
+		if err != nil {
+			return errors.Wrapf(err, "parsing glob filter %q", expr)
+		}
+		f.only = append(f.only, compiledExpr)
+	}
+	f.onlySpec = strings.Join(globExprs, ",")
+	return f.onlyFilterMatchesAnything(cfg)
+}
+
+func (f *NodeGroupFilter) onlyFilterMatchesAnything(cfg *api.ClusterConfig) error {
+	if len(f.only) == 0 {
+		return nil
+	}
+	for i := range cfg.NodeGroups {
+		for _, g := range f.only {
```

Comment:
Can we rename `g` to `nodeGroup` or `ng`?

Reply:
wait, what is `g` here?

Reply:
same for line 85

Reply:
`g` for "glob", will rename to something more readable - thanks!